### PR TITLE
Require exclusive locks for vfs and renames

### PR DESCRIPTION
**Pull request by TheOneRing**
Fixes: #8761, #8754

### CODE REVIEW

File: src/gui/lockwatcher.cpp [hunk 1]
```diff
@@ -52,16 +52,18 @@ void LockWatcher::checkFiles()
 {
     QSet<QString> unlocked;
 
-    foreach (const QString &path, _watchedPaths) {
-        if (!FileSystem::isFileLocked(path)) {
-            qCInfo(lcLockWatcher) << "Lock of" << path << "was released";
-            emit fileUnlocked(path);
-            unlocked.insert(path);
+    for (auto it = _watchedPaths.cbegin(); it != _watchedPaths.constEnd(); ++it) {
```
Comment (dschmidt) [inline src/gui/lockwatcher.cpp:55]: `cbegin` and `constEnd` looks pretty inconsistent

**Review by dschmidt — COMMENTED** [review]
🤷🏻

File: src/gui/lockwatcher.cpp [hunk 2]
```diff
@@ -52,16 +52,18 @@ void LockWatcher::checkFiles()
...
+        if (!FileSystem::isFileLocked(it.key(), it.value())) {
+            qCInfo(lcLockWatcher) << "Lock of" << it.key() << "was released";
+            emit fileUnlocked(it.key());
+            unlocked.insert(it.key());
         }
     }
 
     // Doing it this way instead of with a QMutableSetIterator
     // ensures that calling back into addFile from connected
     // slots isn't a problem.
```
Comment (dragotin) [inline src/gui/lockwatcher.cpp:65]: Not sure if this comment is sufficient still?

**Review by dragotin — APPROVED** [review]
LGTM

**Pull request merged**